pandas/tests/series/test_operators.py

```diff
@@ -1574,28 +1600,28 @@ def test_dt64_series_add_intlike(self, tz):
 
 
 class TestSeriesOperators(TestData):
-    def test_op_method(self):
-        def check(series, other, check_reverse=False):
-            simple_ops = ['add', 'sub', 'mul', 'floordiv', 'truediv', 'pow']
-            if not compat.PY3:
-                simple_ops.append('div')
+    @pytest.mark.parametrize('opname', ['add', 'sub', 'mul', 'floordiv',
+                                        'truediv', 'div', 'pow'])
+    def test_op_method(self, opname):
+        if opname == 'div' and not compat.PY3:
+            pytest.skip('div test only for Py3')
 
-            for opname in simple_ops:
-                op = getattr(Series, opname)
-
-                if op == 'div':
-                    alt = operator.truediv
-                else:
-                    alt = getattr(operator, opname)
+        def check(series, other, check_reverse=False):
+            op = getattr(Series, opname)
 
-                result = op(series, other)
-                expected = alt(series, other)
+            if op == 'div':
+                alt = operator.truediv
```

Comment:
instead of making check an inner function, just add another layer of paramaterization (for self.ts * 2 and self.ts[::2])

Reply:
I think this is incorrect anyway since `op == 'div'` should never be True; this is probably supposed to be `if opname == div`; and in fact that check doesn't matter because we're excluding "div" on PY3 anyway...

Reply:
you would put callables, e.g.
```"ts", [lambda x: x * 2, lambda x: x[::2]]`` and just inject ``self.ts``

e.g.
```
def test_.....(ts):
       ts = ts(self.ts)
 ```

Reply:
This sounds unnecessarily convoluted.  Dummy lambdas and parametrization obfuscate what is being tested (which is already not-immediately-obvious, as you mentioned earlier).

Reply:
this is standard pytest usage. and much much more clear than the current.

Reply:
i'd like this changed